ignition/service/messaging.py

```diff
@@ -218,16 +227,21 @@ def __on_send_error(self, excp):
         logger.error('Error sending envelope', exc_info=excp)
 
     def deliver(self, envelope, key=None):
-        if envelope is None:
-            raise ValueError('An envelope must be passed to deliver a message')
-        self.__lazy_init_producer()
-        content = envelope.message.content
-        logger.debug('Delivering envelope to {0} with message content: {1}'.format(envelope.address, content))
-        if key is None:
-            self.producer.send(envelope.address, content).add_callback(self.__on_send_success).add_errback(self.__on_send_error)
-        else:
-            self.producer.send(envelope.address, key=str.encode(key), value=content).add_callback(self.__on_send_success).add_errback(self.__on_send_error)
-
+        try:
+            if envelope is None:
+                raise ValueError('An envelope must be passed to deliver a message')
+            self.__lazy_init_producer()
+            content = envelope.message.content
+            logger.debug('Delivering envelope to {0} with message content: {1}'.format(envelope.address, content))
+            if key is None:
+                self.producer.send(envelope.address, content).add_callback(self.__on_send_success).add_errback(self.__on_send_error)
+            else:
+                self.producer.send(envelope.address, key=str.encode(key), value=content).add_callback(self.__on_send_success).add_errback(self.__on_send_error)
+            
+            signal(SIGTERM, self.sigterm_handler)
+            
+        except Exception as e:
```

Comment:
Not sure we need this try/except block? We should probably just let the errors be raised without catching them?